AceComm-3.0/ChatThrottleLib.lua

```diff
@@ -292,38 +314,89 @@ end
 -- - ... made up of N "Pipe"s (1 for each destination/pipename)
 -- - and each pipe contains messages
 
+local SendAddonMessageResult = Enum.SendAddonMessageResult or {
+	Success = 0,
+	AddonMessageThrottle = 3,
+	NotInGroup = 5,
+	ChannelThrottle = 8,
+	GeneralError = 9,
+}
+
+local function MapToSendResult(result)
+	if result == true then
+		result = SendAddonMessageResult.Success
+	elseif result == false then
+		result = SendAddonMessageResult.GeneralError
+	end
+
+	return result
+end
+
+local function IsThrottledSendResult(result)
+	return result == SendAddonMessageResult.AddonMessageThrottle
+end
+
+local function PerformSend(sendFunction, ...)
+	-- The select(-1, true, ...) incantation grabs the last return value
+	-- from the called function, defaulting to true if it returned nothing.
+	--
+	-- The default case is required for SendChatMessage. The selection of
+	-- the last return value is required due to per-flavor API divergences.
+
+	bMyTraffic = true
+	local sendResult = select(-1, true, sendFunction(...))
+	bMyTraffic = false
+
+	return MapToSendResult(sendResult)
+end
+
 function ChatThrottleLib:Despool(Prio)
 	local ring = Prio.Ring
 	while ring.pos and Prio.avail > ring.pos[1].nSize do
-		local msg = table_remove(ring.pos, 1)
-		if not ring.pos[1] then  -- did we remove last msg in this pipe?
-			local pipe = Prio.Ring.pos
-			Prio.Ring:Remove(pipe)
-			Prio.ByName[pipe.name] = nil
-			DelPipe(pipe)
-		else
-			Prio.Ring.pos = Prio.Ring.pos.next
-		end
-		local didSend=false
+		local pipe = ring.pos
+		local msg = pipe[1]
+		local sendResult
 		local lowerDest = strlower(msg[3] or "")
+
 		if lowerDest == "raid" and not UnitInRaid("player") then
 			-- do nothing
+			sendResult = SendAddonMessageResult.NotInGroup
 		elseif lowerDest == "party" and not UnitInParty("player") then
 			-- do nothing
+			sendResult = SendAddonMessageResult.NotInGroup
```

Comment:
Incidentally - do we want to delete these checks? The API returns a result code in these cases anyway.

Reply:
If we're sure the chat warning message no longer comes up when attempting, sure.

Reply:
Tested on Retail, Cataclysm and Era, no longer seems to print errors into the chat window. As a side note- INSTANCE_CHAT always succeeds.

As an aside-aside, GUILD still prints "You are not in a guild.", but since we never cared to suppress those, might as well continue to not care.

Reply:
Removed these checks in the squashed version.